File: android/app/src/main/java/com/example/softapplication/Style.kt
package com.example.softapplication

import com.facebook.yoga.YogaAlign
import com.facebook.yoga.YogaEdge
import com.facebook.yoga.YogaFlexDirection
import com.facebook.yoga.YogaJustify
import com.facebook.yoga.YogaNode
import com.facebook.yoga.YogaWrap

class Style {
    companion object {
        fun handleStyle(node: YogaNode, style: Map<String, String>) {
            val display = style["display"]

            node.setMargin(YogaEdge.ALL, 5f)

            for ((key, value) in style) {
                if (key == "display") continue

                when (key) {
                    "flex_direction" -> if (display == "flex") {
                        node.flexDirection = if (value == "column") {
                            YogaFlexDirection.COLUMN
                        } else {
                            YogaFlexDirection.ROW
                        }
                    }

                    "flex_wrap" -> if (display == "flex") {
                        node.wrap = when (value) {
                            "wrap" -> YogaWrap.WRAP
                            "wrap-reverse" -> YogaWrap.WRAP_REVERSE
                            else -> YogaWrap.NO_WRAP
                        }
                    }

                    "flex_grow" -> if (display == "flex") {
                        node.flexGrow = value.toFloat()
                    }

                    "justify_content" -> if (display == "flex") {
                        node.justifyContent = when (value) {
                            "flex-start" -> YogaJustify.FLEX_START
                            "flex-end" -> YogaJustify.FLEX_END
                            "center" -> YogaJustify.CENTER
                            "space-between" -> YogaJustify.SPACE_BETWEEN
                            "space-around" -> YogaJustify.SPACE_AROUND
                            else -> YogaJustify.SPACE_EVENLY
                        }
                    }

                    "align_items" -> if (display == "flex") {
                        node.alignItems = when (value) {
                            "flex-start" -> YogaAlign.FLEX_START
                            "flex-end" -> YogaAlign.FLEX_END
                            "center" -> YogaAlign.CENTER
                            "stretch" -> YogaAlign.STRETCH
                            else -> YogaAlign.BASELINE
                        }
                    }

                    "margin" -> node.setMargin(YogaEdge.ALL, value.toFloat())
                    "margin_top" -> node.setMargin(YogaEdge.TOP, value.toFloat())
                    "margin_right" -> node.setMargin(YogaEdge.RIGHT, value.toFloat())
                    "margin_bottom" -> node.setMargin(YogaEdge.BOTTOM, value.toFloat())
                    "margin_left" -> node.setMargin(YogaEdge.LEFT, value.toFloat())

                    "padding" -> node.setPadding(YogaEdge.ALL, value.toFloat())
                    "padding_top" -> node.setPadding(YogaEdge.TOP, value.toFloat())
                    "padding_right" -> node.setPadding(YogaEdge.RIGHT, value.toFloat())
                    "padding_bottom" -> node.setPadding(YogaEdge.BOTTOM, value.toFloat())
                    "padding_left" -> node.setPadding(YogaEdge.LEFT, value.toFloat())
                }
            }
        }
    }
}

File: src/cpp/private/soft/style.cpp
#include "style.h"

namespace soft::style
{
    void Style::handle_style(YGNodeRef node, std::map<std::string, std::string> styles)
    {        
        std::string display = styles["display"];

        YGNodeStyleSetMargin(node, YGEdgeAll, 5);

        for (auto item : styles)
        {
            std::string key = item.first;

            if (key == "display")
            {
                continue;
            }

            std::string value = item.second;
            
            if (display == "flex")
            {
                if (key == "flex_direction")
                {
                    if (value == "column")
                    {
                        YGNodeStyleSetFlexDirection(node, YGFlexDirectionColumn);
                    }
                    else
                    {
                        YGNodeStyleSetFlexDirection(node, YGFlexDirectionRow);
                    }
                }
                else if (key == "flex_wrap")
                {
                    if (value == "wrap")
                    {
                        YGNodeStyleSetFlexWrap(node, YGWrapWrap);
                    }
                    else if (value == "wrap-reverse")
                    {
                        YGNodeStyleSetFlexWrap(node, YGWrapWrapReverse);
                    }
                    else
                    {
                        YGNodeStyleSetFlexWrap(node, YGWrapNoWrap);
                    }
                }
                else if (key == "flex_grow")
                {
                    YGNodeStyleSetFlexGrow(node, stof(value));
                }
                else if (key == "justify_content")
                {
                    if (value == "flex-start")
                    {
                        YGNodeStyleSetJustifyContent(node, YGJustifyFlexStart);
                    }
                    else if (value == "flex-end")
                    {
                        YGNodeStyleSetJustifyContent(node, YGJustifyFlexEnd);
                    }
                    else if (value == "center")
                    {
                        YGNodeStyleSetJustifyContent(node, YGJustifyCenter);
                    }
                    else if (value == "space-between")
                    {
                        YGNodeStyleSetJustifyContent(node, YGJustifySpaceBetween);
                    }
                    else if (value == "space-around")
                    {
                        YGNodeStyleSetJustifyContent(node, YGJustifySpaceAround);
                    }
                    else
                    {
                        YGNodeStyleSetJustifyContent(node, YGJustifySpaceEvenly);
                    }
                }
                else if (key == "align_items")
                {
                    if (value == "flex-start")
                    {
                        YGNodeStyleSetAlignItems(node, YGAlignFlexStart);
                    }
                    else if (value == "flex-end")
                    {
                        YGNodeStyleSetAlignItems(node, YGAlignFlexEnd);
                    }
                    else if (value == "center")
                    {
                        YGNodeStyleSetAlignItems(node, YGAlignCenter);
                    }
                    else if (value == "stretch")
                    {
                        YGNodeStyleSetAlignItems(node, YGAlignStretch);
                    }
                    else
                    {
                        YGNodeStyleSetAlignItems(node, YGAlignBaseline);
                    }
                }
            }
            else if (key == "margin")
            {
                YGNodeStyleSetMargin(node, YGEdgeAll, stof(value));
            }
            else if (key == "margin_top")
            {
                YGNodeStyleSetMargin(node, YGEdgeTop, stof(value));
            }
            else if (key == "margin_right")
            {
                YGNodeStyleSetMargin(node, YGEdgeRight, stof(value));
            }
            else if (key == "margin_bottom")
            {
                YGNodeStyleSetMargin(node, YGEdgeBottom, stof(value));
            }
            else if (key == "margin_left")
            {
                YGNodeStyleSetMargin(node, YGEdgeLeft, stof(value));
            }
            else if (key == "padding")
            {
                YGNodeStyleSetPadding(node, YGEdgeAll, stof(value));
            }
            else if (key == "padding_top")
            {
                YGNodeStyleSetPadding(node, YGEdgeTop, stof(value));
            }
            else if (key == "padding_right")
            {
                YGNodeStyleSetPadding(node, YGEdgeRight, stof(value));
            }
            else if (key == "padding_bottom")
            {
                YGNodeStyleSetPadding(node, YGEdgeBottom, stof(value));
            }
            else if (key == "padding_left")
            {
                YGNodeStyleSetPadding(node, YGEdgeLeft, stof(value));
            }
        }
    }
}